--- a/package/utils/busybox/Config-defaults.in.orig	2014-08-22 12:12:22.027097538 +0800
+++ b/package/utils/busybox/Config-defaults.in	2014-08-22 12:14:18.723095187 +0800
@@ -231,7 +231,7 @@
 	default n
 config BUSYBOX_DEFAULT_FEATURE_EDITING_HISTORY
 	int
-	default 256
+	default 30 
 config BUSYBOX_DEFAULT_FEATURE_EDITING_SAVEHISTORY
 	bool
 	default n
@@ -792,7 +792,7 @@
 	default n
 config BUSYBOX_DEFAULT_CLEAR
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_DEALLOCVT
 	bool
 	default n
@@ -1158,10 +1158,10 @@
 	default n
 config BUSYBOX_DEFAULT_GETTY
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_LOGIN
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_LOGIN_SESSION_AS_CHILD
 	bool
 	default n
@@ -1833,7 +1833,7 @@
 	default n
 config BUSYBOX_DEFAULT_RX
 	bool
-	default n
+	default y
 config BUSYBOX_DEFAULT_SETSID
 	bool
 	default n
@@ -2145,19 +2145,19 @@
 	default n
 config BUSYBOX_DEFAULT_TELNET
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_TELNET_TTYPE
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_TELNET_AUTOLOGIN
 	bool
 	default n
 config BUSYBOX_DEFAULT_TELNETD
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_TELNETD_STANDALONE
 	bool
-	default y
+	default n
 config BUSYBOX_DEFAULT_FEATURE_TELNETD_INETD_WAIT
 	bool
 	default n
